p2p/host/routed/routed.go

```diff
@@ -107,7 +107,39 @@ func (rh *RoutedHost) Connect(ctx context.Context, pi peer.AddrInfo) error {
 
 	// if we're here, we got some addrs. let's use our wrapped host to connect.
 	pi.Addrs = addrs
-	return rh.host.Connect(ctx, pi)
+	err := rh.host.Connect(ctx, pi)
+	if err != nil {
+		// We couldn't connect. Let's check if we have the most
+		// up-to-date addresses for the given peer. If there
+		// are addresses we didn't know about previously, we
+		// try to connect again.
+		newAddrs, addrErr := rh.findPeerAddrs(ctx, pi.ID)
+		if addrErr != nil {
+			return fmt.Errorf("%s: %s", err, addrErr)
```

Comment:
This error message doesn't make a lot of sense. What about this:
```go
return fmt.Errorf("failed to find peers: %w", addrErr)
```

You can then also rename the `addrErr` to `err`.

Reply:
done